content/en/blog/2025/infernece-extension-support/index.md

```diff
@@ -0,0 +1,155 @@
+
+## **Gateway API Inference Extension: A Kubernetes-Native Solution**
+
+The [Kubernetes Gateway API Inference Extension](https://gateway-api-inference-extension.sigs.k8s.io/guides/#__tabbed_3_2) has introduced solutions to these challenges, building on the proven foundation of Kubernetes Gateway API while adding AI-specific intelligence. Rather than requiring organizations to patch together custom solutions or abandon their existing Kubernetes infrastructure, the extension provides a standardized, vendor-neutral approach to intelligent AI traffic management.
+
+The extension introduces two key Custom Resource Definitions that work together to address the routing challenges we've outlined. The InferenceModel resource provides an abstraction for AI-Inference workload owners to define logical model endpoints, while the InferencePool resource gives platform operators the tools to manage backend infrastructure with AI workload awareness.
+
+By extending the familiar Gateway API model rather than creating an entirely new paradigm, the inference extension enables organizations to leverage their existing Kubernetes expertise while gaining the specialized capabilities that AI workloads demand. This approach ensures that teams can adopt intelligent inference routing aligned with familiar networking knowledge and tooling.
+
+Note: InferenceModel is likely to change in future Gateway API Inference Extension releases.
+
+### **InferenceModel**
+
+The InferenceModel resource allows inference workload owners to define logical model endpoints that abstract the complexities of backend deployment.
+
+{{< text yaml >}}
+apiVersion: inference.networking.x-k8s.io/v1alpha2
+kind: InferenceModel
+metadata:
+  name: customer-support-bot
+  namespace: ai-workloads
+spec:
+  modelName: customer-support
+  criticality: Critical
+  poolRef:
+    name: llama-pool
+  targetModels:
+    - name: llama-3-8b-customer-v1
+      weight: 80
+    - name: llama-3-8b-customer-v2
+      weight: 20
+{{< /text >}}
+
+This configuration exposes a customer-support model that intelligently routes between two backend variants, enabling safe rollouts of new model versions while maintaining service availability.
+
+### **InferencePool**
+
+The InferencePool acts as a specialized backend service that understands AI workload characteristics:
+
+{{< text yaml >}}
+apiVersion: inference.networking.x-k8s.io/v1alpha2
+kind: InferencePool
+metadata:
+  name: llama-pool
+  namespace: ai-workloads
+spec:
+  targetPortNumber: 8000
+  selector:
+    app: llama-server
+    version: v1
+  extensionRef:
+    name: llama-endpoint-picker
+{{< /text >}}
+
+When integrated with Istio, this pool automatically discovers model servers through the istio’s service discovery.
```

Comment:
```suggestion
When integrated with Istio, this pool automatically discovers model servers through service discovery.
```

(Fixing the typo, but calling out also that I don't understand the sentence.  It discovers model servers... based on their labels? Or are you about to tell me how in the next section?)

Reply:
no, based on istio's service discovery. We deploy a Service behind the scene and gets all of istio's semantics of creating endpoints and networking config from a Service

Reply:
then "Istio's service discovery", but not "the istio's"